packages/agreement-process/test/agreementService.integration.test.ts

```diff
@@ -169,6 +172,42 @@ const expectedAgreementCreation = async (
   return actualAgreement;
 };
 
+const expectedAgreementArchived = async (
```

Comment:
I know this function is to replicate what we did for agreement creation - however, in this case, it is only used once so I would just remove the function and do that directly in the test (as in https://github.com/pagopa/interop-be-monorepo/pull/380 or https://github.com/pagopa/interop-be-monorepo/pull/384).

In general, I avoid writing generalizations/utility functions when testing unless they provide a lot of value - they add more code to maintain, more logic that can make tests fail, and it is harder to know if failures are due to a business logic bug or to a logic bug in test utilities.